lib/vtls/rustls.c

```diff
@@ -0,0 +1,409 @@
+/***************************************************************************
+ *                                  _   _ ____  _
+ *  Project                     ___| | | |  _ \| |
+ *                             / __| | | | |_) | |
+ *                            | (__| |_| |  _ <| |___
+ *                             \___|\___/|_| \_\_____|
+ *
+ * Copyright (C) 2020, Jacob Hoffman-Andrews, <github@hoffman-andrews.com>
+ *
+ * This software is licensed as described in the file COPYING, which
+ * you should have received as part of this distribution. The terms
+ * are also available at https://curl.se/docs/copyright.html.
+ *
+ * You may opt to use, copy, modify, merge, publish, distribute and/or sell
+ * copies of the Software, and permit persons to whom the Software is
+ * furnished to do so, under the terms of the COPYING file.
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTY OF ANY
+ * KIND, either express or implied.
+ *
+ ***************************************************************************/
+#include "curl_setup.h"
+
+#ifdef USE_RUSTLS
+
+#include "curl_printf.h"
+
+#include <errno.h>
+#include <crustls.h>
+
+#include "urldata.h"
+#include "sendf.h"
+#include "vtls.h"
+
+struct ssl_backend_data
+{
+  struct rustls_client_session *session;
+  bool data_pending;
+};
+
+static const struct rustls_client_config *client_config = NULL;
+
+static int
+Curl_rustls_init(void)
+{
+  client_config = rustls_client_config_new();
+  return 1;
+}
+
+static size_t
+Curl_rustls_version(char *buffer, size_t size)
+{
+  return msnprintf(buffer, size, "rustls");
```

Comment:
Is there really no version number? I would strongly urge you to provide one, as this is often key material when debugging issues and tracking down suspicious behavior in user reports.